analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/Checks/ConstantCheck.cs

```diff
@@ -21,29 +21,35 @@
 using Microsoft.CodeAnalysis;
 using SonarAnalyzer.Helpers;
 using SonarAnalyzer.SymbolicExecution.Constraints;
+using StyleCop.Analyzers.Lightup;
 
 namespace SonarAnalyzer.SymbolicExecution.Roslyn.Checks
 {
     internal class ConstantCheck : SymbolicCheck
     {
         protected override ProgramState PreProcessSimple(SymbolicContext context) =>
-            ConstraintFromValue(context.Operation.Instance.ConstantValue.Value) is { } constraint
-                ? context.SetOperationConstraint(constraint)
+            context.Operation.Instance.ConstantValue.HasValue
+            && ConstraintFromConstantValue(context.Operation) is { } value
+                ? context.State.SetOperationValue(context.Operation, value)
                 : context.State;
 
         public static SymbolicConstraint ConstraintFromType(ITypeSymbol type) =>
-            ConstraintFromValue(DefaultValue(type));
+            type.Is(KnownType.System_Boolean) ? BoolConstraint.False : null;
 
-        public static SymbolicConstraint ConstraintFromValue(object value) =>
-            value switch
-            {
-                // Update DefaultValue when adding new types
-                true => BoolConstraint.True,
-                false => BoolConstraint.False,
-                _ => null
-            };
+        public static SymbolicValue ConstraintFromConstantValue(IOperationWrapperSonar operation) =>
```

Comment:
```suggestion
        private static SymbolicValue ConstraintFromConstantValue(IOperationWrapperSonar operation) =>
```
If this is possible, the `operation.Instance.ConstantValue.HasValue` check can be removed.

Reply:
We have to keep it, because we started matching `null` now. And `HasValue=true + null` is different than `HasValue=false + null` (effectively any other `IOperation`)

Reply:
Ah, it's private and tested at entry point. So indeed, we don't need it